quaistats/quaistats.go

```diff
@@ -190,28 +224,54 @@ func parseEthstatsURL(url string) (parts []string, err error) {
 }
 
 // New returns a monitoring service ready for stats reporting.
-func New(node *node.Node, backend backend, engine consensus.Engine, url string) error {
+func New(node *node.Node, backend backend, engine consensus.Engine, url string, trustednode bool) error {
 	parts, err := parseEthstatsURL(url)
 	if err != nil {
 		return err
 	}
 
-	tpsLookupCache, _ := lru.New(c_tpsLookupCacheLimit)
-	gasLookupCache, _ := lru.New(c_gasLookupCacheLimit)
+	chainID := backend.ChainConfig().ChainID
+	var durationLimit *big.Int
+
+	switch {
+	case chainID.Cmp(chainID9000) == 0:
+		durationLimit = params.DurationLimit
```

Comment:
you couldn't get the duration limit directly from the consensus config ?, this is fine for now, but would be nice to not have this logic here, we cannot track the Chain ID changes for different networks in different places. Please make an issue to fix it

Reply:
https://github.com/dominant-strategies/go-quai/issues/1268

Reply:
Likely will need to add that to consensus config. I couldn't find where it was located and ended up down several rabbit holes